### Задача 1
*Когда используется критерий Стьюдента, а когда Z –критерий?*  

Самое главное отличие - это известна ли нам дисперсия генеральной совокупности - это $Z$-криттерий или нет ,тогда используем криттерий Стьюдента $t$ .  
При этом в обоих случаях должно соблюдаться условие нормальности распределения генеральной совокупности ,откуда берутся выборки, и равенство дисперсий между тестируемыми выборками нашей случайной величины.

---

(Задачи 2,3 решать вручную)

### Задача 2
*Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.*

Дано:
* $\mu_0$ = 17 мм
* $\bar{X}$ = 17.5 мм
* $\alpha$ = 0.05
* $n$ = 100
* $\sigma^2$ =4 мм^2(следовательно, $\sigma= \sqrt{4}=2$)


Решение:  
Так как дисперсия нам известна, будем использовать Z -криттерий:  
при нулевой гипотезе $\mu=\mu_0=17$ , его будем подставлять в формулу ниже
$$Z_н=\frac{\bar{X} - \mu}{\sigma/ \sqrt{n}}=\frac{17.5 - 17}{2/ \sqrt{100}}=\frac{0.5}{0.2}= 2.5$$
За альтернативную гипотезу тогда принимаем, что $\mu>\mu_0$  

Теперь сравним нашу оценку отклонения с отклонениям по статистической значимости $\alpha$ , для этого найдем $Z_{\alpha}$ по таблице накопленной вероятности:  
$Z_{\alpha} = 1.645 $, ниже которого мы попадаем в верные(для гипотезы $H_0$) значения  (левее $\alpha$ на графике).  
Так как $Z_н >Z_{\alpha}$ делаем вывод, что верна альтернативная гипотеза, то есть **истинный средний диаметр подшипников будет больше заявленного производителем**

In [15]:
import scipy.stats as st

st.norm.ppf(0.95)

1.6448536269514722

### Задача 3
*Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:*

    202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
    
*Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)*

Дано:
* $\mu_0$ = 200 г
* $\sigma$ неизвестна, следовательно, используем криттерий Стьюдента
* $\alpha$ = 0.01 ,двусторонний криттерий будет $\pm t_{0.995}$ 
* $n$ = 10


Решение:  
Нулевая гипотеза: $\mu=\mu_0=200$  
Альтернативная гипотеза: $\mu \neq \mu_0$

Среднее по выборке рассчитаем по данному нам списку значений, $\bar{X}=\frac{\Sigma x}{n}$ = 198.5 г  
Дисперсию мы тоже можем посчитать по значениям $\sigma_н^2 = \frac{\Sigma(X-\bar{X})^2}{n-1} = 17.85$  
Тогда мы можем рассчитать криттерий Стьюдента по формуле:  
$$t_н=\frac{\bar{X} - \mu}{\sigma_н/ \sqrt{n}}=\frac{198.5 - 200}{\sqrt{17.85}/ \sqrt{10}}=\frac{-1.5}{4.22*3.16}= -0.112$$


Табличное значение криттерия Стьюдента для двустороннней $\alpha=0.001$ смотрим по степени свободы $k=9$ и вероятности $p=0.995$
$$t_{\frac{\alpha}{2}}=\pm 3.25$$
Так как $t_н<-t_{\frac{\alpha}{2}}$ , то делаем вывод о верности нулевой гипотезы, то есть **утверждение продавца** про среднее по весу в 200 г **верно**.

In [27]:

import numpy as np
cookies_w= np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
X= np.mean(cookies_w)
X



198.5

In [28]:
disp = np.var(cookies_w)
disp

17.85

In [29]:
t_found=(X-200)/(np.std(cookies_w)/np.sqrt(len(cookies_w)))
t_found

-1.1227217828476797

Проверю через Питон вывод:

In [32]:
#в эту функцию мы передаем выборку, среднее для нулевой гипотезы, и вариант альтернативной гипотезы. two-sided по умолчанию
#здесь также есть ключевой аргумент, который указывает что делать с nan значениями - считать как NaN->Nan, выдавать ошибку или игнорить в расчетах
st.ttest_1samp(cookies_w, 200 , alternative='two-sided')

Ttest_1sampResult(statistic=-1.0651074037450894, pvalue=0.31456660352472854)

Видим, что двусторонний p-value=0.31 больше нашей  $\frac{\alpha}{2}=0.005$, значит **верна нулевая гипотеза**. При этом стоит обратить внимание, что расчетный криттерий statistic у нас немного отличается , -1.06 против нашего -1.12

### Задача 4
(решать с помощью функции) *Есть ли статистически значимые различия в росте дочерей?*

    Рост матерей:  [172, 177, 158, 170, 178,175, 164, 160, 169, 165]
    Рост взрослых дочерей: [173, 175, 162, 174, 175, 168, 155, 170, 160]

Если говорят о матерях и дочерях, то выборки, скорее всего зависимые, но у нас при этом не совпадает количество элементов. Простой человеческий фактор предполагает, что нужно убрать одно лишнее значение с конца списка матерей. Дальше решаем по двухвыборочному тесту в  Питоне

In [ ]:
mothers =   [172, 177, 158, 170, 178,175, 164, 160, 169]
daughters = [173, 175, 162, 174, 175, 168, 155, 170, 160]
st.ttest_rel(mothers,daughters)